packages/widgets/src/tabbar.ts

```diff
@@ -711,6 +712,44 @@ export class TabBar<T> extends Widget {
    * Handle the `'keydown'` event for the tab bar.
    */
   private _evtKeyDown(event: KeyboardEvent): void {
+    // Allow for navigation using tab key
+    if (event.key === 'Tab') {
+      return;
+    }
+
+    // Check if Enter or Spacebar key has been pressed and open that tab
+    if (
+      event.key === 'Enter' ||
+      event.key === 'Spacebar' ||
+      event.key === ' '
+    ) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      // Get focus element that is in focus by the tab key
+      let focusedElement = document.activeElement;
+
+      if (focusedElement) {
+        if (focusedElement.className === 'lm-TabBar-addButton') {
```

Comment:
This should check if the class name is on the list of classes as there can be more than one class (in particular downstream can add other classes).